File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.go
func sumAndMultiply(n int) int64 {
	if n == 0 {
		return 0
	}

	count := int(math.Log10(math.Abs(float64(n)))) + 1
	pow := int(math.Pow(10, float64(count-1)))

	var x int64 = 0
	var sum int64 = 0

	for pow > 0 {
		a := int64(n / pow)

		n %= pow
		pow /= 10

        if a == 0 {
            continue
        }

		x = a + x*10
		sum += a
	}

	return sum * x
}

File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        long x = 0;
        long sum = 0;
        int count = (n == 0) ? 1 : (int) Math.log10(Math.abs(n)) + 1;
        long pow = (long) Math.pow(10, count-1);

        while (n > 0) {
            long a = n / pow;
            if (a == 0) {
                n %= pow;
                pow /= 10;
                continue;
            }
            x = a + x * 10;
            sum += a;
            n %= pow;
            pow /= 10;
        }

        return sum * x;
    }
}